File: PC/remote/protocol.py
"""
Up0k Remote
Protocol Utilities
"""

from __future__ import annotations

import json


def encode_packet(data: dict) -> bytes:
    """Convert a dictionary into bytes."""

    return json.dumps(data).encode("utf-8")


def decode_packet(data: bytes) -> dict:
    """Convert bytes into a dictionary."""

    return json.loads(data.decode("utf-8"))

File: Android/app/src/main/java/com/up0k/remote/network/tcp/Protocol.kt
package com.up0k.remote.network.tcp

import org.json.JSONObject

/**
 * Up0k Remote
 * Network Protocol
 */

object Protocol {

    fun encode(packet: JSONObject): String {
        return packet.toString()
    }

    fun decode(data: String): JSONObject {
        return JSONObject(data)
    }

}